pipelines/pro_image.go

```diff
@@ -0,0 +1,62 @@
+package pipelines
+
+import (
+	"context"
+	"fmt"
+
+	"dagger.io/dagger"
+	"github.com/grafana/grafana-build/containers"
+)
+
+func ProImage(ctx context.Context, dc *dagger.Client, directory *dagger.Directory, args PipelineArgs) error {
+	debianPackageFile := dc.Host().Directory("./").File(args.ProImageOpts.Deb)
+
+	hostedGrafanaRepo, err := containers.CloneWithGitHubToken(dc, args.ProImageOpts.GithubToken, "https://github.com/grafana/hosted-grafana", "main")
+	if err != nil {
+		return fmt.Errorf("cloning hosted-grafana repo: %w", err)
+	}
+
+	socketPath := "/var/run/docker.sock"
+	socket := dc.Host().UnixSocket(socketPath)
+
+	container := dc.Container().From("golang:1.20-alpine").
+		WithUnixSocket(socketPath, socket).
+		WithExec([]string{"apk", "add", "--update", "make", "docker", "git", "jq", "jsonnet"}).
+		WithDirectory("/src", hostedGrafanaRepo).
+		WithFile("/src/grafana.deb", debianPackageFile, dagger.ContainerWithFileOpts{}).
+		WithWorkdir("/src").
+		WithExec([]string{
+			"/bin/sh", "-c",
+			"docker build --platform=linux/amd64 . -f ./cmd/hgrun/Dockerfile -t hgrun:latest",
+		}).
+		WithExec([]string{
+			"/bin/sh", "-c",
+			fmt.Sprintf("docker build --platform=linux/amd64 --build-arg=RELEASE_TYPE=%s --build-arg=GRAFANA_VERSION=%s --build-arg=HGRUN_IMAGE=hgrun:latest . -f ./docker/hosted-grafana-all/Dockerfile -t hostedgrafana:latest",
+				args.ProImageOpts.ReleaseType,
+				args.ProImageOpts.GrafanaVersion,
+			),
+		})
+
+	if args.ProImageOpts.Push {
+		panic("TODO: push to container registry")
```

Comment:
Here you'll need to set up an Auth type, similar to this:

https://github.com/grafana/grafana-build/blob/main/containers/google_cloud.go#L113

and call its Authenticate function on a container to activate-service-account. Using the Auth struct with the Authenticate function it will automatically attempt to use your $HOME/.gcloud directory if you don't specify a service-account.